bin/add-test-widgets-to-sidebar.php

```diff
@@ -0,0 +1,460 @@
+			'settings' => array_merge(
+				amp_media_widget( 'video' ),
+				array(
+					'title'   => 'Test Video Widget: With Loop, Preload',
+					'loop'    => true,
+					'preload' => 'metadata',
+				)
+			),
+		),
+	);
+}
+
+/**
+ * Get the settings for a media widget.
+ *
+ * Can apply to 'audio' or 'video' widgets.
+ * Queries for a single post of the media type.
+ * If none exists, amp_media() will throw an error.
+ *
+ * @param  string $type The type of media. like 'audio' or 'video'.
+ * @return array $audio_widget The settings for the media widget.
+ */
+function amp_media_widget( $type ) {
+	$all_media = amp_media( $type, 1 );
+	$media     = reset( $all_media );
+	return array(
+		'attachment_id' => $media->ID,
+		'url'           => $media->guid,
+	);
+}
+
+/**
+ * Get the settings for the gallery widget.
+ *
+ * @return array $audio_widget The settings for the gallery widget.
+ */
+function amp_gallery_widget() {
+	$images = amp_media( 'image', 3 );
+	return array(
+		'ids' => wp_list_pluck( $images, 'ID' ),
+	);
+}
+
+/**
+ * Gets the settings for the image widget.
+ *
+ * @return array $image_widget The settings for the image widget.
+ */
+function amp_image_widget() {
+	$all_images        = amp_media( 'image', 1 );
+	$image             = reset( $all_images );
+	$metadata          = wp_get_attachment_metadata( $image->ID );
+	$default_dimension = 100;
+	return array(
+		'attachment_id' => $image->ID,
+		'height'        => isset( $metadata['height'] ) ? $metadata['height'] : $default_dimension,
+		'width'         => isset( $metadata['width'] ) ? $metadata['width'] : $default_dimension,
+		'url'           => wp_get_attachment_url( $image->ID ),
+	);
+}
+
+/**
+ * Get media items of a certain type.
+ *
+ * @throws Exception If not enough posts exist.
+ * @param  integer $type  The post_mime_type of the media item.
+ * @param  integer $count The number of images for which to query.
+ * @return array|WP_CLI::error The media IDs, or an error on failure.
+ */
+function amp_media( $type, $count = 3 ) {
+	$query = new \WP_Query(
+		array(
+			'post_type'      => 'attachment',
+			'post_mime_type' => $type,
+			'post_status'    => 'inherit',
+			'posts_per_page' => $count,
+		)
+	);
+	if ( $query->post_count < $count ) {
+		throw new Exception(
+			sprintf(
+				'Please ensure at least %1$s "%2$s" attachments are accessible and run this again. There are currently only %3$s.',
+				$count,
+				$type,
+				$query->found_posts
+			)
+		);
+	}
+	return $query->get_posts();
+}
+
+/**
+ * Gets a menu ID, if it has more than 4 items.
+ *
+ * Iterates through all of the menus.
+ * And returns the first that has 4 items.
+ *
+ * @throws Exception When there is no menu with 4 items.
+ * @return integer|WP_CLI::error $menu_id The menu ID, or an error if no menu has at least 4 items.
+ */
+function amp_menu() {
+	$menus         = wp_get_nav_menus();
+	$minimum_count = 4;
+	foreach ( $menus as $menu ) {
+		if ( $menu->count > $minimum_count ) {
```

Comment:
This actually requires 5 menu items.

Reply:
Fixed [here](https://github.com/Automattic/amp-wp/pull/859/commits/dba75caef1efc20d9f9e88888069f11ead68bce8).